### PR TITLE
Remove `in_band_lifetimes` for `rustc_passes`

### PR DESCRIPTION
**Pull request by pitaj**
#91867

### CODE REVIEW

File: compiler/rustc_passes/src/check_attr.rs [hunk 1]
```diff
@@ -382,7 +382,7 @@ impl CheckAttrVisitor<'tcx> {
         &self,
         hir_id: HirId,
         attr_span: &Span,
-        attrs: &'hir [Attribute],
+        attrs: &'_ [Attribute],
```
Comment (michaelwoerister) [inline compiler/rustc_passes/src/check_attr.rs:385]: Isn't this the same as diagnostics `attrs: &[Attribute]`?

File: compiler/rustc_passes/src/check_attr.rs [hunk 2]
```diff
@@ -1663,7 +1663,7 @@ impl CheckAttrVisitor<'tcx> {
         }
     }
 
-    fn check_used(&self, attrs: &'hir [Attribute], target: Target) {
+    fn check_used(&self, attrs: &'_ [Attribute], target: Target) {
```
Comment (michaelwoerister) [inline compiler/rustc_passes/src/check_attr.rs:1666]: Same here.

File: compiler/rustc_passes/src/check_attr.rs [hunk 3]
```diff
@@ -1481,7 +1481,7 @@ impl CheckAttrVisitor<'tcx> {
     /// Checks if the `#[repr]` attributes on `item` are valid.
     fn check_repr(
         &self,
-        attrs: &'hir [Attribute],
+        attrs: &'_ [Attribute],
```
Comment (michaelwoerister) [inline compiler/rustc_passes/src/check_attr.rs:1484]: Same here.

**Review by michaelwoerister — COMMENTED** [review]
Thanks for the PR, @pitaj!

This looks good to me, except for the nit about using `&'_ [Attribute]` instead of just `&[Attribute]`.

**Comment by pitaj** [comment]
@michaelwoerister good catch, those should be fixed now.

**Comment by michaelwoerister** [comment]
Thanks! I'll r+ after the required rebase.

**Comment by pitaj** [comment]
@michaelwoerister rebased and ready

**Comment by michaelwoerister** [comment]
Thanks, @pitaj!

@bors r+ rollup

**Pull request merged**